src/Reflection/Adapter/ReflectionNamedType.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection\Adapter;
+
+class ReflectionNamedType extends ReflectionType
+{
+    public function getName()
+    {
+        return $this->betterReflectionType->getName();
```

Comment:
Maybe we should implement `getName` on the parent type, then let inheritance do its thing here?

Reply:
We don't need the `ReflectionNamedType` at all after that change... Hmm?

Reply:
True that, but it would still be there for symmetry with the core API

Reply:
But there's no `getName()` method in `ReflectionType` in the core API :) How much symmetric should we be? :)

So maybe it would be better to just fix the `protected`?

Reply:
I'm fine with protected if it's `@internal` then 👍

Reply:
So anything I should do now? :)

Reply:
Either we mark the `protected` field as `@internal`, or we declare `getName()` on `ReflectionType`. I think the first approach is cleaner, to avoid polluting the API

Reply:
I've made a version where everything is private. Is it ok or should I change it to `@internal`?

Reply:
All good as-is :+1: